File: 0007-reverse-integer/0007-reverse-integer.ts
function reverse(x: number): number {
  const sign = x < 0 ? -1 : 1;
  let ans = 0;
  x = Math.abs(x);
  while (x) {
    ans = ans * 10 + x % 10;
    x = x / 10 | 0;
  }
  return ans <= 2 ** 31 ? ans * sign : 0;
};

File: 0007-reverse-integer/0007-reverse-integer.go
func reverse(x int) int {
  sign, ans := 1, 0
  if x < 0 {
    sign = -1
    x = -x
  }
  for x != 0 {
    ans = ans * 10 + x % 10
    x /= 10
  }
  ans *= sign
  if ans < -(1 << 31) || ans > (1 << 31) - 1 {
    return 0
  }
  return ans
}